travis/setup_python.sh

```diff
@@ -0,0 +1,42 @@
+#!/bin/bash -ex
+
+# Script to set up Python using native platform tools rather than conda
+
+echo "==================== Starting executing ci-helpers scripts ====================="
+
+if [[ -z $PYTHON_VERSION ]]; then
+    echo "$PYTHON_VERSION needs to be set";
+    exit 1;
+fi
+
+if [[ $TRAVIS_OS_NAME == windows ]]; then
+    choco install --no-progress python --version $PYTHON_VERSION;
+    export PATH="/c/Python37:/c/Python37/Scripts:$PATH"
+    python3 -m venv ~/python;
+    source ~/python/bin/activate;
+    python -m pip install --upgrade pip;
+fi
+
+if [[ $TRAVIS_OS_NAME == osx ]]; then
+
+    if [[ $PYTHON_VERSION == 3.6 ]]; then
+        FULL_PYTHON_VERSION=3.6.9;
+    elif [[ $PYTHON_VERSION == 3.7 ]]; then
+        FULL_PYTHON_VERSION=3.7.9;
+    elif [[ $PYTHON_VERSION == 3.8 ]]; then
+        FULL_PYTHON_VERSION=3.8.6;
+    elif [[ $PYTHON_VERSION == 3.9 ]]; then
+        FULL_PYTHON_VERSION=3.9.0;
```

Comment:
is the plan here to regularly revisit these numbers and update?

Reply:
I can write a small script to determine these automatically so we don't have to, but didn't want to spend time on it if you disagreed with it overall.

Reply:
(It's OK if it's, but then maybe it's time to keep a tally of such updatable stuff (the current ones I know about are: latest versions of hard wired things in `LATEST...`, this python version, conda version)

Reply:
I think it's reasonable enough to come back to it regularly, less sudden surprises that way (similar as with the conda verison)